### PR TITLE
force valid file path for asset file mapping operations

### PR DESCRIPTION
**Pull request by birarda**
- fix and issue where assets could be uploaded with invalid folder paths

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Plan

This fixes an issue where assets could be uploaded to the Asset Server with a path that specified a folder and not a file.

1. Attempt to upload an asset at an empty path (clear the input when asked for the asset mapping). This upload should fail.
2. Attempt to upload an asset at a folder path with a trailing slash (like `/my/cool/folder/`). This upload should fail.

Both of the above uploads succeed on master.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7aac2e69167718f44bd1b1f64c28fd9acc3a28f1 available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-7aac2e6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-7aac2e6.dmg

File: assignment-client/src/assets/AssetServer.cpp [hunk 1]
```diff
@@ -637,7 +637,7 @@ bool AssetServer::renameMapping(AssetPath oldPath, AssetPath newPath) {
     oldPath = oldPath.trimmed();
     newPath = newPath.trimmed();
 
-    if (!isValidPath(oldPath) || !isValidPath(newPath)) {
+    if (!isValidFilePath(oldPath) || !isValidFilePath(newPath)) {
         qWarning() << "Cannot perform rename with invalid paths - both should have leading forward slashes:"
```
Comment (huffman) [inline assignment-client/src/assets/AssetServer.cpp]: This should be updated to say "leading and ending forward slashes"

File: libraries/networking/src/AssetUtils.h [hunk 2]
```diff
@@ -31,7 +31,8 @@ const size_t SHA256_HASH_LENGTH = 32;
 const size_t SHA256_HASH_HEX_LENGTH = 64;
 const uint64_t MAX_UPLOAD_SIZE = 1000 * 1000 * 1000; // 1GB
 
-const QString ASSET_PATH_REGEX_STRING = "^\\/(?!\\/)(?:[^\\/]|\\/(?!\\/))*$";
+const QString ASSET_FILE_PATH_REGEX_STRING = "^\\/([^\\/\\0]+(\\/)?)*([^\\/\\0]+)$";
```
Comment (huffman) [inline libraries/networking/src/AssetUtils.h]: Can this be reduced to `^(\\/[^\\/\\0])+$`?

Comment (birarda) [inline libraries/networking/src/AssetUtils.h]: Close but you're basically right - `^(\\/[^\\/\\0]+)+$`

Comment (huffman) [inline libraries/networking/src/AssetUtils.h]: Yeah, one character folder/filenames wouldn't be very good :P

**Comment by hifi-gustavo** [comment]
Build console output for commit: 76f65ae4d7bfe8376b7a01922655db263429ca28 available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-76f65ae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-76f65ae.dmg

File: assignment-client/src/assets/AssetServer.cpp (hunk 1 above)
Comment (birarda) [inline assignment-client/src/assets/AssetServer.cpp]: It should say "leading forward and no ending slashes"

**Comment by hifi-gustavo** [comment]
Build console output for commit: b80f3fbd7bbd3fbc3818c98e5452f891cea03576 available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-b80f3fb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-b80f3fb.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 179cb67cdb5e5895b735a85b3c34afdf1c506655 available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-179cb67.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-179cb67.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eb8f67b612b470bb145d9b7999a70cf79849b9b6 available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-eb8f67b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-eb8f67b.dmg

File: assignment-client/src/assets/AssetServer.cpp (hunk 1 above)
Comment (huffman) [inline assignment-client/src/assets/AssetServer.cpp]: Yes, you are right, apparently I can't get any of my comments right today

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3f194b4dd1e76c7525d18567c9b72c2dbb08dfb available at: https://gist.github.com/8f456e739efdefb5d5225fe9e405448a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-e3f194b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8855-e3f194b.dmg

**Comment by Nex-Pro** [comment]
### PASS

No longer able to create a blank line at the top of the ATP list.  Thanks @birarda

**Pull request merged**